### DISCUSSION TITLE
Using recommendFit feature to fetch a list of proposals

### DISCUSSION THREAD

**Discussion by Racem-s** [TimefoldAI/timefold-solver#552] [General]
Hello everyone

I am new to Timefold AI, I already asked my first question on [stackoverflow](https://stackoverflow.com/questions/77700655/is-it-possible-to-get-multi-solutions-in-timefold)
@ge0ffrey's response allowed me to progress, but I still have some problems

To resume, I have a list of delivery persons, my clients called me and I should give them some proposals to delivery their order on live.

Using recommendFit feature, I succeed to have some proposals (one proposal shift by a delivery person), but it's not sufficient: 

Let's imagine that we have 2 delivery persons. Both of them work from 8 am until 6 pm and I would like to propose to my client a delivery in the next 2 days. The delivery duration is dynamic, I calculate it based on locations of clients and the delivery person, let's say that for our example the delivery take one hour.

1. The first delivery person, take his lunch from 12 until 13h

- First day: has already a delivery from 8 am to 10 am and another one from 4pm until 5pm => so I should propose a delivery at the first day at 10am (just after his first delivery) or at 1 pm ((just after his lunch)  or at 5pm (just after his seconddelivery) 

- second day: there is no scheduled delivery => I can propose a delivery at 8 am or at 1 pm

2. The second delivery person take his lunch from 1pm until 2pm

- First day he is not available from 8 am to 9 => I can propose to my client a delivery at 9am or at 2pm
- Second day: there is no scheduled delivery => I can propose a delivery at 8 am or at 2 pm

Is Timefold can help in this case ?

**Comment by ge0ffrey** [comment]
Related to https://github.com/TimefoldAI/timefold-quickstarts/issues/235

**Comment by ge0ffrey** [comment]
This sounds a very similar to recommendTimeWindow for vehicle routing, which is a for loop and some code around recommendFit calls. There is no public example of that yet.

Basically, the request sends in a visit to schedule and a list of time windows. For each time window, the SolutionManager.recommendFit() is called for that visit, with the visit.minStart and visit.maxEnd (ready & due time) confined to the time window. Each top recommendation goes into a result map with the time window as key. That map is returned from the function.

**Reply by Racem-s** [reply]
Thanks alot @ge0ffrey for your response, well appreciated
I look forward your use-case using recommendFit() feature ! 
at the same time, I started using it in my MVP, As there are not enough examples, I move forward slowly :)
